File: app/src/androidTest/java/de/westnordost/streetcomplete/quests/AddSportTest.java
package de.westnordost.streetcomplete.quests;

import java.util.ArrayList;

import de.westnordost.streetcomplete.data.osm.OsmElementQuestType;
import de.westnordost.streetcomplete.data.osm.changes.StringMapEntryAdd;
import de.westnordost.streetcomplete.data.osm.changes.StringMapEntryModify;
import de.westnordost.streetcomplete.quests.sport.AddSport;
import de.westnordost.streetcomplete.quests.sport.AddSportForm;

public class AddSportTest extends AOsmElementQuestTypeTest
{
	@Override public void setUp() throws Exception
	{
		super.setUp();
		tags.put("leisure","pitch");
	}

	public void testReplaceHockey()
	{
		tags.put("sport","hockey");
		bundle.putStringArrayList(AddSportForm.OSM_VALUES, getAsStringArray("field_hockey"));
		verify(new StringMapEntryModify("sport","hockey","field_hockey"));
	}

	public void testReplaceTeamHandball()
	{
		tags.put("sport","team_handball");
		bundle.putStringArrayList(AddSportForm.OSM_VALUES, getAsStringArray("handball"));
		verify(new StringMapEntryModify("sport","team_handball","handball"));
	}

	public void testAddSport()
	{
		bundle.putStringArrayList(AddSportForm.OSM_VALUES, getAsStringArray("soccer"));
		verify(new StringMapEntryAdd("sport","soccer"));
	}

	private ArrayList<String> getAsStringArray(String value)
	{
		ArrayList<String> values = new ArrayList<>();
		values.add(value);
		return values;
	}

	@Override protected OsmElementQuestType createQuestType()
	{
		return new AddSport(null);
	}
}

File: app/src/androidTest/java/de/westnordost/streetcomplete/quests/AddSportTest.kt
package de.westnordost.streetcomplete.quests

import de.westnordost.streetcomplete.data.osm.changes.StringMapEntryAdd
import de.westnordost.streetcomplete.data.osm.changes.StringMapEntryModify
import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
import de.westnordost.streetcomplete.quests.sport.AddSport
import org.junit.Test

import org.mockito.Mockito.mock

class AddSportTest : AOsmElementQuestTypeTest() {

    override val questType = AddSport(mock(OverpassMapDataDao::class.java))

    @Test fun replaceHockey() {
        tags["sport"] = "hockey"
        bundle.putStringArrayList(ImageListQuestAnswerFragment.OSM_VALUES, arrayListOf("field_hockey"))
        verify(StringMapEntryModify("sport", "hockey", "field_hockey"))
    }

    @Test fun replaceTeamHandball() {
        tags["sport"] = "team_handball"
        bundle.putStringArrayList(ImageListQuestAnswerFragment.OSM_VALUES, arrayListOf("handball"))
        verify(StringMapEntryModify("sport", "team_handball", "handball"))
    }

    @Test fun replaceSkating() {
        tags["sport"] = "skating"
        bundle.putStringArrayList(ImageListQuestAnswerFragment.OSM_VALUES, arrayListOf("ice_skating"))
        verify(StringMapEntryModify("sport", "skating", "ice_skating"))
    }

    @Test fun replaceFootball() {
        tags["sport"] = "football"
        bundle.putStringArrayList(ImageListQuestAnswerFragment.OSM_VALUES, arrayListOf("soccer"))
        verify(StringMapEntryModify("sport", "football", "soccer"))
    }

    @Test fun addSport() {
        bundle.putStringArrayList(ImageListQuestAnswerFragment.OSM_VALUES, arrayListOf("soccer"))
        verify(StringMapEntryAdd("sport", "soccer"))
    }
}
